dnsflow_read.py

```diff
@@ -1,125 +1,188 @@
 #!/usr/bin/env python
 
-'''
+"""
 See dnsflow.c header comment for packet formats.
-'''
 
-import sys, time, argparse
+Utility functions to simplify interface.
+"""
+import time
+import argparse
+import gzip
 import socket
-import dpkt, pcap
+import dpkt
 import struct
+import sys
 import ipaddr
+from dpkt.ip import IP_PROTO_UDP
+from dpkt.udp import UDP
+
+try:
+    import pcap
+except ImportError:
+    print(
+        "Could not import the 'pcap' module.\n"
+        "Please see README for dependency instructions and ensure pypcap (not python-libpcap) is installed."
+    )
+    sys.exit(-1)
```

Comment:
@jfalkner-nokia There wasn't a very helpful way from the module attributes to distinguish between the two modules. However in the python3 environment the `import pcap` should fail if the old python-libpcap is installed, but the new (pypcap) module has not been installed (at least that's what happened for me locally). I suppose it is possible that some customer currently using python2 (<= 4.2) would want to newly install this, but the code now doesn't support this case. I don't get the sense this tool is installed on very many customers anyways -- I couldn't find many at least. I also made an extra note in the README about the package names. I'm calling it good enough for now, I want to get back to monolith :upside_down_face:

Reply:
All sounds good to me. Thanks for checking it out